### PR TITLE
Deprecate minthrottle to use motor_idle

### PR DESCRIPTION
**Pull request by haslinghuis**
- keep in sync with firmware https://github.com/betaflight/betaflight/pull/13936
- deprecates `minthrottle`
- show `motor_idle` for every configured protocol.
- updates tooltip
- change precision

### CODE REVIEW

**Comment by nerdCopter** [comment]
when MULTISHOT, PWM, PROSHOT, i don't see minThrottle and do see maxThrottle, minCommand
image

**Comment by haslinghuis** [comment]
Correct `minthrottle` is no longer used in firmware (when merged).

**Comment by nerdCopter** [comment]
and successfully backwards compatible for <=4.5
image

**Review by nerdCopter — APPROVED** [review]
- approving for workflow not to be merged before firmware PR

File: locales/en/messages.json [hunk 1]
```diff
@@ -1464,10 +1464,10 @@
...
+    "configurationmotorIdle": {
         "message": "Motor Idle (%)"
     },
-    "configurationDigitalIdlePercentHelp": {
+    "configurationmotorIdleHelp": {
         "message": "The Motor Idle value sets the idle speed of the motors when throttle is at minimum position.<br/><br/><strong>Dynamic Idle disabled</strong><br/><br/>The lowest throttle value sent to any motor, while armed, as a percentage of full throttle. Increase it to improve motor startup reliability, to avoid desyncs, and to improve PID responsiveness at low throttle.<br/><br/>Too low: the motors may not start up reliably, or desync at the end of a flip or roll or on hard throttle chops.<br/><br/>Too high: the craft may feel 'floaty'.<br/><br/><strong>Dynamic idle enabled</strong><br/><br/>The maximum throttle allowed, after arming, before takeoff. If RPM is less than dyn_idle_min_rpm, or zero, this throttle value will be sent to the motors. When the motors start spinning, throttle is adjusted to maintain the set RPM, but cannot exceed this value until the quad takes off.<br/><br/>Too low: the motors may not start up reliably.<br/><br/>Too high: the craft may shake on the ground before takeoff."
     },
```
Comment (ctzsnooze) [inline locales/en/messages.json:1472]: maybe 
`The Motor Idle value sets the idle speed of the motors when throttle is 'low' (i.e, when the throttle setpoint is below min_check).` 
`If Dynamic Idle is not enabled, the value sent to the motors will be 'min_command' plus the Motor Idle percentage, for both PWM and DShot ESCs.`
Too low: the motors may not start up reliably, or desync at the end of a flip or roll or on hard throttle chops.
Too high: the craft may feel 'floaty'.  < no changes for these two lines>

`If Dynamic Idle is enabled, and before takeoff, the motor drive value cannot be greater than the Motor Idle percentage above 'min_command'. If the motors do not start up reliably, increase the Motor Idle value to send a higher initial startup value. Once the motors start spinning, Dynamic idle will attempt to attain the set minimum RPM. Note that, before takeoff, the amount of drive is lower at idle than in-flight, due to the Motor Idle limit, and the RPM at idle may be below the configured dyn_idle_min_rpm value. So long as the motors start up reliably, this is OK. To check that your configured dyn_idle_min_rpm value will be achieved in flight, raise throttle above the 'air mode_start_throttle' value, which overcomes the Motor Idle limit.  If the Motor Idle is set too high, and Dynamic Idle is enabled, the craft may shake on the ground before takeoff.`

Sorry for so many words !

**Comment by ctzsnooze** [comment]
@haslinghuis thanks for this work - I can confirm @nerdCopter's test results for 4.6, where selecting an analog protocol still shows minThrottle, where it should show the same MotorIdle value as we see with DShot:

Screen Shot 2024-10-05 at 11 02 57 am

**Comment by haslinghuis** [comment]
On 4.6 we have:

image

On 4.5 we have

image

File: locales/en/messages.json (hunk 1 above)
Comment (haslinghuis) [reply locales/en/messages.json:1472]: Is this the longest tooltip :stuck_out_tongue: 

image

Comment (ctzsnooze) [reply locales/en/messages.json:1472]: OK let's try something shorter :-)
```
Sets the idle speed of the motors while armed and throttle is 'low' (below `min_check`).
**Dynamic Idle not enabled:**
Each motor gets `min_command` plus Motor Idle percent.
Idle too low: motors may not start reliably, spin up slowly, or desync in flips or rolls.
Idle too high: the craft may feel 'floaty'.
Note: analog ESC's must be calibrated so the motors start just above `min_command`.
**Dynamic Idle enabled**
On arming, the 'normal' idle value is sent to each motor, until they spin up.
Once spinning, the motor signal is adjusted to achieve the target RPM.
Before takeoff, the motor signal is limited to the Motor Idle percentage, and the set RPM may not be achieved.  This is OK.  When throttle is raised above the `airmode_motor_start_throttle` percentage, the limit is much higher, and the set RPM should be achieved at idle.
Idle too low: motors may not start reliably
Idle too high: shaking before takeoff (only if dynamic idle is also high)
Note: Dynamic Idle requires DShot and DShot Telemetry.

Comment (ctzsnooze) [reply locales/en/messages.json:1472]: We need the tooltips to be wider :-)
Either that, or the tooltip is responsive to the state of Dynamic idle, and only shows the text for 'not enabled' when it is 'not enabled', and vice versa.
This is perhaps the best solution.

**Comment by ctzsnooze** [comment]
Approving - motor idle renders as it should in 4.6.  Let's see if we can make the message itself sensitive to the dynamic idle state, that should make it small enough.
Maybe these tooltips can auto-size themselves a bit wider for computer screens?

**Review by ctzsnooze — APPROVED** [review]

**Comment by haslinghuis** [comment]
- Have tried two messages, it updates in dev console, but does not update GUI (like done in other places).
- Using wide tip we get:
image

**Comment by ctzsnooze** [comment]
Wide tip looks good!  Maybe update the text with the shorter version that I provided, and merge?

**Comment by haslinghuis** [comment]
image

**Comment by haslinghuis** [comment]
image

**Comment by ctzsnooze** [comment]
@haslinghuis - while we are thinking about this, my personal view is that the user doesn't need two decimal points of accuracy in the idle percentage, even though we provide it with the 16 bit value in the CLI.  
Maybe we can in Configurator only provide 1 decimal point of adjustment?  ie 5.1, 5.2 etc steps, not 5.10, 5.11, 5.12, when using the up and down arrows?

**Pull request merged**